File: flut/.flutter/lib/flutter/gestures/gesture_settings.dart
import 'package:flutter/gestures.dart';
import 'package:flut/flut/runtime.dart';
import 'package:flut/flut/object.dart';

class FlutDeviceGestureSettings extends FlutValueObject {
  final DeviceGestureSettings settings;
  const FlutDeviceGestureSettings(this.settings)
    : super('DeviceGestureSettings');

  @override
  Map<String, dynamic> flutEncode() {
    final result = flutBaseProps();
    if (settings.touchSlop != null) {
      result['touchSlop'] = settings.touchSlop;
    }
    return result;
  }

  static DeviceGestureSettings? flutDecode(
    FlutRuntime runtime,
    Map<String, dynamic> data,
  ) {
    return DeviceGestureSettings(
      touchSlop: runtime.unpackOptionalField<double>(data, 'touchSlop'),
    );
  }
}

File: flut/flutter/gestures/gesture_settings.py
from typing import Optional, override

from flut._flut.engine import FlutValueObject, _flut_pack_value
from flut._flut.runtime import _flut_unpack_optional_field


class DeviceGestureSettings(FlutValueObject):
    _flut_type = "DeviceGestureSettings"

    def __init__(self, *, touchSlop: Optional[float] = None):
        super().__init__()
        self.touchSlop = touchSlop

    @property
    def panSlop(self) -> Optional[float]:
        return self.touchSlop * 2.0 if self.touchSlop is not None else None

    @staticmethod
    def _flut_unpack(data: dict) -> "DeviceGestureSettings":
        return DeviceGestureSettings(
            touchSlop=_flut_unpack_optional_field(data, "touchSlop"),
        )

    @override
    def _flut_pack(self) -> dict:
        result = self._flut_base_props()
        if self.touchSlop is not None:
            result["touchSlop"] = _flut_pack_value(self.touchSlop)
        return result
